File: CS/DevExpress.Example01/SearchBehavior/SearchBehavior.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Grid;
using System.Windows;
using System.Windows.Input;

namespace DevExpress.Example01.SearchBehavior {
    
    public class SearchBehavior : Behavior<GridControl> {

        protected SearchWindow _SearchWindow;

        #region Commands

        #region SearchCommand

        protected DelegateCommand _Search;

        public DelegateCommand Search {
            get {
                if(this._Search == null) {
                    this._Search = new DelegateCommand(this.SearchExecute);
                }

                return this._Search;
            }
        }

        protected void SearchExecute() {
            if(this._SearchWindow == null) {
                this._SearchWindow = new SearchWindow(this.AssociatedObject);
                this._SearchWindow.Owner = Application.Current.MainWindow;

            }

            this._SearchWindow.Show();
        }

        #endregion SearchCommand

        #region HideSearch

        protected DelegateCommand _HideSearch;

        public DelegateCommand HideSearch {
            get {
                if(this._HideSearch == null) {
                    this._HideSearch = new DelegateCommand(this.HideSearchExecute);
                }

                return this._HideSearch;
            }
        }

        protected void HideSearchExecute() {
            if(this._SearchWindow != null) {
                this._SearchWindow.Close();
            }
        }

        #endregion HideSearch
 
        #endregion Commands

        protected override void OnAttached() {
            base.OnAttached();
            this.AssociatedObject.InputBindings.Add(new KeyBinding(this.Search, new KeyGesture(Key.F, ModifierKeys.Control)));
            this.AssociatedObject.InputBindings.Add(new KeyBinding(this.HideSearch, new KeyGesture(Key.Escape)));
        }

    }
}

File: VB/DevExpress.Example01/SearchBehavior/SearchBehavior.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Grid
Imports System.Windows
Imports System.Windows.Input

Namespace DevExpress.Example01.SearchBehavior

    Public Class SearchBehavior
        Inherits Behavior(Of GridControl)

        Protected _SearchWindow As SearchWindow

#Region "Commands"
#Region "SearchCommand"
        Protected _Search As DelegateCommand

        Public ReadOnly Property Search As DelegateCommand
            Get
                If _Search Is Nothing Then
                    _Search = New DelegateCommand(AddressOf SearchExecute)
                End If

                Return _Search
            End Get
        End Property

        Protected Sub SearchExecute()
            If _SearchWindow Is Nothing Then
                _SearchWindow = New SearchWindow(AssociatedObject)
                _SearchWindow.Owner = Application.Current.MainWindow
            End If

            _SearchWindow.Show()
        End Sub

#End Region  ' SearchCommand
#Region "HideSearch"
        Protected _HideSearch As DelegateCommand

        Public ReadOnly Property HideSearch As DelegateCommand
            Get
                If _HideSearch Is Nothing Then
                    _HideSearch = New DelegateCommand(AddressOf HideSearchExecute)
                End If

                Return _HideSearch
            End Get
        End Property

        Protected Sub HideSearchExecute()
            If _SearchWindow IsNot Nothing Then
                _SearchWindow.Close()
            End If
        End Sub

#End Region  ' HideSearch
#End Region  ' Commands
        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AssociatedObject.InputBindings.Add(New KeyBinding(Search, New KeyGesture(Key.F, ModifierKeys.Control)))
            AssociatedObject.InputBindings.Add(New KeyBinding(HideSearch, New KeyGesture(Key.Escape)))
        End Sub
    End Class
End Namespace
